Description: When "defaultdomain" is null, don't add an FQDN to hostnames.
Author: Nick Leverton <nick@leverton.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=504184

Index: nullmailer-1.10/lib/hostname.cc
===================================================================
--- nullmailer-1.10.orig/lib/hostname.cc	2012-05-16 08:03:37.000000000 +0100
+++ nullmailer-1.10/lib/hostname.cc	2012-05-16 08:32:04.000000000 +0100
@@ -33,12 +33,16 @@
 {
   int nome;
   nome = 0;
-  if (!config_read("me", me)) {
+  // introduced as bufix for #120660, #157259, #158412 in 1.00RC5-17;
+  // still there since it's more appropriate for Debian systems
+  if (!config_read("../mailname", me)) {
     nome = 1;
     me = "me";
   }
-  if (!config_read("defaultdomain", defaultdomain))
-    defaultdomain = nome ? "defaultdomain" : me.c_str();
+  // allow defaultdomain to be empty for Debian
+  config_read("defaultdomain", defaultdomain);
+  // if (!config_read("defaultdomain", defaultdomain))
+    // defaultdomain = nome ? "defaultdomain" : me.c_str();
   if (!config_read("defaulthost", defaulthost))
     defaulthost = nome ? "defaulthost" : me.c_str();
   canonicalize(defaulthost);
Index: nullmailer-1.10/src/queue.cc
===================================================================
--- nullmailer-1.10.orig/src/queue.cc	2012-05-16 08:03:41.000000000 +0100
+++ nullmailer-1.10/src/queue.cc	2012-05-16 08:03:42.000000000 +0100
@@ -85,8 +85,8 @@
     if(hostname == me || hostname == "localhost")
       addr = adminaddr;
   }
-  else if(hostname.find_first('.') < 0)
-    return false;
+  // else if(hostname.find_first('.') < 0)
+    // return false;
   return true;
 }
 
